utils/lldp.go

```diff
@@ -0,0 +1,179 @@
+)
+
+// Implement utilities to fetch lldp information aobut connecting neighbors.
+// The lldp information can be used in non cloud environment to automate
+// networking configuration on the physical devices
+
+// LLDPNeighbor lists the neighbor attributes as learned by the host
+type LLDPNeighbor struct {
+	SystemName        string
+	SystemDescription string
+	PortID            string
+	PortDesc          string
+	CiscoACIPodID     string
+	CiscoACINodeID    string
+}
+
+// FetchLLDPInfo reads the lldp information using lldptool
+// A native go lib could be done but it requires that lldpad daemon is
+// aceessible natively on the lldp ipc path
+func FetchLLDPInfo(ifname string) (*LLDPNeighbor, error) {
+	lldp := &LLDPNeighbor{}
+
+	output, err := exec.Command("docker", "exec", "lldpad", "lldptool",
+		"-n", "-t", "-i", ifname).CombinedOutput()
+	if err != nil {
+		log.Errorf("Error reading lldp information. Error: %s Output: \n%s\n", err, output)
+		return nil, err
+	}
+
+	err = parseLLDPArgs(string(output), lldp)
+
+	return lldp, err
+}
+
+func parseLLDPArgs(lldpToolOutput string, lldp *LLDPNeighbor) error {
+	var err error
+	lines := strings.Split(lldpToolOutput, "\n")
+	maxIdx := len(lines) - 1
+	for idx := 0; idx < maxIdx; {
+		line := strings.Trim(lines[idx], "\n")
+		idx++
+
+		if !strings.Contains(line, "TLV") ||
+			strings.Contains(line, "End of LLDPDU TLV") {
+			continue
+		}
+
+		attrIdx := idx
+		for !strings.Contains(lines[attrIdx], "TLV") && attrIdx <= maxIdx {
+			lines[attrIdx] = strings.Trim(lines[attrIdx], "\n")
+			attrIdx++
+		}
+		tlvLines := lines[idx:attrIdx]
+		idx = attrIdx
+		switch {
+		case strings.Contains(line, "System Name TLV"):
+			lldp.SystemName, err = lldpReadSystemName(tlvLines)
+		case strings.Contains(line, "System Description TLV"):
+			lldp.SystemDescription, err = lldpReadSystemDescription(tlvLines)
+		case strings.Contains(line, "Port ID TLV"):
+			lldp.PortID, err = lldpReadPortID(tlvLines)
+		case strings.Contains(line, "Port Description TLV"):
+			lldp.PortDesc, err = lldpReadPortDescription(tlvLines)
+		default:
+			continue
+		}
+		if err != nil {
+			log.Errorf("Error parsing lldp information. Error '%v' \n", err)
+		}
+	}
+	log.Infof("VJ before lldpInfo = %#v \n", lldp)
+	err = deriveLLDPArgs(lldp)
+	if err != nil {
+		log.Errorf("Error parsing lldp information. Error '%v' \n", err)
+	}
+	log.Infof("VJ after lldpInfo = %#v \n", lldp)
+
+	return err
+}
+
+func deriveLLDPArgs(lldp *LLDPNeighbor) error {
+	var err error
+
+	lldp.CiscoACIPodID, err = lldpReadACIPodID(lldp.SystemDescription)
+	if err != nil {
+		log.Errorf("Error parsing lldp information. Error '%v' \n", err)
+	}
+
+	lldp.CiscoACINodeID, err = lldpReadACINodeID(lldp.SystemDescription)
+	if err != nil {
+		log.Errorf("Error parsing lldp information. Error '%v' \n", err)
+	}
+
+	return err
+}
+
+func lldpReadSystemName(attrLines []string) (string, error) {
+	if len(attrLines) < 1 {
+		return "", errors.New("empty attributes")
```

Comment:
suggest to use `core.Error` package here and elsewhere as that is used across this project.

Also you can define a variable for the `empty attributes` and use it elsewhere below.